### PR TITLE
nox: add missing check for sync in dual view

### PR DESCRIPTION
**Pull request by ndellingwood**
Address issue #6790

 Changes to be committed:
	modified:   test/tpetra/ME_Tpetra_1DFEM_def.hpp

@trilinos/nox 

## Motivation

Address issue #6790 

## Testing

Reproduced the failure in test `NOX_Tpetra_1DFEM_MPI_4` in a host build, confirmed it passes with these changes.

### CODE REVIEW

File: packages/nox/test/tpetra/ME_Tpetra_1DFEM_def.hpp [hunk 1]
```diff
@@ -325,13 +325,19 @@ evalModelImpl(const Thyra::ModelEvaluatorBase::InArgs<Scalar> &inArgs,
     typedef typename tpetra_vec::execution_space execution_space;
 
     // Create ghosted objects
+    // Should there be braces corresponding to this if-statement and the following lines of code (as in xPtr_ case below)?
```
Comment (ndellingwood) [inline packages/nox/test/tpetra/ME_Tpetra_1DFEM_def.hpp]: @rppawlo can you confirm if braces should wrap the code after the if-statement, similar to the `xPtr_` case below?

**Comment by ndellingwood** [comment]
Marked WIP until confirmation regarding potential missing braces for an if-statement.

File: packages/nox/test/tpetra/ME_Tpetra_1DFEM_def.hpp [hunk 2]
```diff
@@ -325,13 +325,19 @@ evalModelImpl(const Thyra::ModelEvaluatorBase::InArgs<Scalar> &inArgs,
...
     if (is_null(uPtr_))
       uPtr_ = Teuchos::rcp(new tpetra_vec(xGhostedMap_));
+
+      if (uPtr_->need_sync_host())
+        uPtr_->sync_host();
       uPtr_->modify_host();
```
Comment (mhoemmen) [inline packages/nox/test/tpetra/ME_Tpetra_1DFEM_def.hpp]: @rppawlo Why are you calling `modify_host` on the target of an Import?  It looks like this was intended to force MultiVector to pack and unpack on host.  That doesn't necessarily work like you think it does any more.  For example, the target of an Export or Import has the right to control its own sync state.

File: packages/nox/test/tpetra/ME_Tpetra_1DFEM_def.hpp (hunk 1 above)
Comment (rppawlo) [reply packages/nox/test/tpetra/ME_Tpetra_1DFEM_def.hpp]: @ndellingwood - the uPtr code should NOT be wrapped in braces. Each iteration we will need a new import on the solution vector (uPtr). The xPtr holds nodal coords and is not changing from iteration to iteration so only needs to be called once. We can probably remove the modify_host() calls as @mhoemmen suggested.

File: packages/nox/test/tpetra/ME_Tpetra_1DFEM_def.hpp (hunk 2 above)
Comment (rppawlo) [reply packages/nox/test/tpetra/ME_Tpetra_1DFEM_def.hpp]: @mhoemmen - not sure. This was written by someone who has left sandia, so I can't get a quick answer. Can we just remove the modify calls?

Comment (mhoemmen) [reply packages/nox/test/tpetra/ME_Tpetra_1DFEM_def.hpp]: @rppawlo We should remove them, unless downstream code implicitly depends on MultiVector being sync'd to host.  There's no good way to test for that, though.

Comment (ndellingwood) [reply packages/nox/test/tpetra/ME_Tpetra_1DFEM_def.hpp]: > We should remove them, unless downstream code implicitly depends on MultiVector being sync'd to host

@rppawlo @mhoemmen I'll hold off making any changes until getting a definite "yes" to remove the `modify_host` calls (and the sync checking I added in this PR)

File: packages/nox/test/tpetra/ME_Tpetra_1DFEM_def.hpp (hunk 1 above)
Comment (ndellingwood) [reply packages/nox/test/tpetra/ME_Tpetra_1DFEM_def.hpp]: @rppawlo thanks for clarifcation.

File: packages/nox/test/tpetra/ME_Tpetra_1DFEM_def.hpp (hunk 2 above)
Comment (rppawlo) [reply packages/nox/test/tpetra/ME_Tpetra_1DFEM_def.hpp]: yes - let's remove them.

**Comment by ndellingwood** [comment]
@rppawlo @mhoemmen I removed the sync changes I introduced, removed the unnecessary `modify_host` calls, and changed indentation of the uPtr_ import call to prevent potential warnings about misleading indentation. Rebased the changes for a clean single commit

**Comment by mhoemmen** [comment]
Thanks @ndellingwood ! :-D

**Pull request merged**